
Agresti: Ch 3 # 13,14,15

### 13. Access the horseshoe crab data of Table 3.2 at www.stat.ufl.edu/∼aa/intro-cda/appendix.html.

a. Using x = weight and Y = number of satellites, fit a Poisson loglinear model. Report the prediction equation.

logμˆ = ˆα + βxˆ =  -0.4284+0.0006x     (g)

logμˆ = ˆα + βxˆ =  -0.4284+0.59x     (kg)


In [1]:
library(dplyr)
HorseshoeCrab<-read.csv("HorseshoeCrab.csv")
colnames(HorseshoeCrab)
HorseshoeCrab<-HorseshoeCrab%>%mutate(weight=weight/1000)
head(HorseshoeCrab)
mdl.pois <- glm(satellites~weight,family=poisson(link="log"),data=HorseshoeCrab)
summary(mdl.pois)


Attaching package: ‘dplyr’

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



[1] "color"      "spine"      "width"      "satellites" "weight"    
[6] "Y"

color spine width satellites weight Y
1 3     3     28.3  8          3.05   1
2 4     3     22.5  0          1.55   0
3 2     1     26.0  9          2.30   1
4 4     3     24.8  0          2.10   0
5 4     3     26.0  4          2.60   1
6 3     3     23.8  0          2.10   0


Call:
glm(formula = satellites ~ weight, family = poisson(link = "log"), 
    data = HorseshoeCrab)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.9307  -1.9981  -0.5627   0.9298   4.9992  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept) -0.42841    0.17893  -2.394   0.0167 *  
weight       0.58930    0.06502   9.064   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 632.79  on 172  degrees of freedom
Residual deviance: 560.87  on 171  degrees of freedom
AIC: 920.16

Number of Fisher Scoring iterations: 5


b. Estimate the mean of Y for female crabs of average weight 2.44 kg.

The model fit yields an estimated mean number of 2.75 (~3)  satellites μˆ, a fitted value, at 2440 g (i.e. 2.44kg)

In [2]:
fitted_value_2.44kg<-exp( -0.4284+(0.59*2.44))
fitted_value_2.44kg

[1] 2.748898

c. Use βˆ to describe the weight effect. Construct a 95% confidence interval for β and for the multiplicative effect of a 1 kg increase.

logμˆ = ˆα + βxˆ =  -0.4284+0.59x     (kg)

βˆ in this model is additive, not multiplicative.

A 1 kg increase in weight has a predicted increase of βˆ= 0.59 in the expected number of sattelites.

The multiplicative effect on the fitted value for each of 1 unit increase in weight (kg) is exp(.59)= 1.803

For the Poisson model, with weight in kg, the 95% Wald confidence interval for the effect of weight (β) is 0.59 ± 1.96(0.07), which is (0.45, 0.73).

For the Poisson model, the 95% Wald confidence interval for the multiplicative effect of weight in kg (β) is exp(0.59 ± 1.96(0.07)), which is (1.57, 2.08)

In [3]:
exp(.59)

#95% confidence interval for βˆ 
round(0.59-(1.96*(0.07)),4)
round(0.59+(1.96*(0.07)),4)


#95% confidence interval for multiplicative effect of a 1 kg increase.
exp(round(0.59-(1.96*(0.07)),2))
exp(round(0.59+(1.96*(0.07)),2))

[1] 1.803988

[1] 0.4528

[1] 0.7272

[1] 1.568312

[1] 2.075081

d. Conduct a Wald test of the hypothesis that the mean of Y is independent of weight. Interpret.

H0: β = 0 against Ha: β $\neq$ 0 

z = β/SE

The Z score 9.64 has a p value less than .05 which shows that Y is dependent on weight (kg).

In [4]:
0.58930/0.06502
2*pnorm(0.58930/0.06502,lower.tail=F) # two tail test
pnorm(9.64) #left tail
1- pnorm(9.64) #upper  right tail

[1] 9.063365

[1] 1.264868e-19

[1] 1

[1] 0

e. Conduct a likelihood-ratio test about the weight effect. Interpret.

The LR chisquare statistic is 71.925, df=1 with a corresponding p value (0.00) less than .05 which provides a strong evidence of a weight effect (kg) on the mean number of satellites. 

*Notice that the p value for the LR chisquare statistic is the same as the one for the Wald test.

In [5]:
library(package=car)
Anova(mdl.pois,test="LR")
pchisq(mdl.pois$null.deviance-mdl.pois$deviance,1,lower.tail=F)

Loading required package: carData

Attaching package: ‘car’

The following object is masked from ‘package:dplyr’:

    recode



LR Chisq Df Pr(>Chisq)  
weight 71.92524 1  2.235066e-17

[1] 2.235066e-17

### 14. Refer to the previous exercise. Allow overdispersion by fitting the negative binomial loglinear model.

In [6]:
library(dplyr)
library(MASS)
HorseshoeCrab<-read.csv("HorseshoeCrab.csv")
colnames(HorseshoeCrab)
HorseshoeCrab<-HorseshoeCrab%>%mutate(weight=weight/1000)
mdl.nb <- glm.nb(satellites~weight, data = HorseshoeCrab)
summary(mdl.nb)


Attaching package: ‘MASS’

The following object is masked from ‘package:dplyr’:

    select



[1] "color"      "spine"      "width"      "satellites" "weight"    
[6] "Y"


Call:
glm.nb(formula = satellites ~ weight, data = HorseshoeCrab, init.theta = 0.9310592338, 
    link = log)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.8394  -1.4122  -0.3247   0.4744   2.1279  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  -0.8647     0.4048  -2.136   0.0327 *  
weight        0.7603     0.1578   4.817 1.45e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for Negative Binomial(0.9311) family taken to be 1)

    Null deviance: 216.43  on 172  degrees of freedom
Residual deviance: 196.16  on 171  degrees of freedom
AIC: 754.64

Number of Fisher Scoring iterations: 1


              Theta:  0.931 
          Std. Err.:  0.168 

 2 x log-likelihood:  -748.644 

a. Report the prediction equation and the estimate of the dispersion parameter and its SE. Is there evidence that this model gives a better fit than the
Poisson model?


The predicted equation is log(μ)ˆ = −0.8647 + 0.7603x

Dˆ=0.9311
SE= 0.1578

There is evidence that the NB model gives a better fit because we can see that the Dispersion paramater is significant such that the value is greater than 0, which indicates overdispersion such that the variance is greater than the mean. 

b. Construct a 95% confidence interval for β. Compare it with the one in (c) in the previous exercise. Interpret, and explain why the interval is wider with the negative binomial model.

For the Negative bionomial regression  model, with weight in kg, the 95% Wald confidence interval for the effect of weight (β)  is (-1.174, -0.5554) 

The interval is wider with the negative binomial model because it is conservative and it allows overdispersion ie it allows the variance to exceed the mean.  


In [7]:
#95% confidence interval for Negative Binomial Regression Model
round(-0.8647 -(1.96*(0.1578)),4)
round(-0.8647 +(1.96*(0.1578)),4)

[1] -1.174

[1] -0.5554

### 15. A recent General Social Survey asked subjects, “Within the past 12 months, how many people have you known personally that were victims of homicide”? The sample mean for the 159 blacks was 0.522, with a variance of 1.150. The sample mean for the 1149 whites was 0.092, with a variance of 0.155.

a. Let yij denote the response for subject j of race i, and let μij = E(Yij ).
The Poisson model log(μij ) = α + βxij with x1j = 1 (blacks) and x2j =
0 (whites) has fit log(μˆij ) = −2.38 + 1.733xij . Show that the estimated
population means are 0.522 for blacks and 0.092 for whites, which are the
sample means.


The Poisson model log(μij ) = α + βxij 

log(μˆij ) = −2.38 + 1.733xij

x1j = 1 (blacks)

x2j = 0 (whites)

Code can be show below that the the estimated population means are 0.522 for blacks and 0.092 for whites, which are the sample means.

In [8]:
exp(-2.38+1.733) # x1j = 1 (blacks)
exp(-2.38) #x2j = 0 (whites)

[1] 0.5236143

[1] 0.09255058

b. For the Poisson GLM, the standard error of βˆ is 0.147. Show that the Wald
95% confidence interval for the ratio of means for blacks and whites is
(4.2, 7.5). [Hint: Note that β is the log of the ratio of the means.]

Code can be show below that the Wald 95% confidence interval for the ratio of means for blacks and whites is
(4.2, 7.5)

In [9]:
round(exp(1.733-(1.96*.147)),1)
round(exp(1.733+(1.96*0.147)),1)

[1] 4.2

[1] 7.5

c. The negative binomial loglinear model has the same estimates as in (a), but the standard error of βˆ increases to 0.238 and the Wald 95% confidence interval for the ratio of means is (3.5, 9.0). Based on the sample means and variances, which confidence interval is more believeable? Why?

The sample mean for the 159 blacks was 0.522, with a variance of 1.150. The sample mean for the 1149 whites was 0.092, with a variance of 0.155.

For Poisson distributions, the variance equals the mean.
The (negative binomial loglinear model) Wald 95% confidence interval for the ratio of means is more believable because as we can see for the sample means and variances of the data, the variances are greater than the mean. The negative binomial distribution allows the variance to exceed the mean. 

d. The negative binomial model has Dˆ = 4.94 (SE = 1.00). Explain why this shows strong evidence that the negative binomial GLM is more appropriate than the Poisson GLM.

This shows strong evidence that the negative binomial GLM is more appropriate than the Poisson GLM because the estimate of D (dispersion parameter) is greater than 0, indicating greater disperson and therefore, the variance is greater than the mean, suggesting that the Poisson model is not appropriate for this data.  
